frontend/components/BlockyBoard.tsx

```diff
@@ -242,12 +259,12 @@ export default function BlockyBoard() {
       boardCanvas.removeEventListener('contextmenu', contextMenuListener);
       // boardCanvas.removeEventListener('mousemove', mouseMoveListener);
     };
-  }, [dimensions]);
+  }, [mergedProps]);
```

Comment:
The useEffect cleanup function above should properly dispose of the game by unregistering all events and disabling the timer. The `window` keylistener should also be removed.